File: sum.py
'''
$ time python sum.py 
5000000050000000

real	0m10.930s
user	0m4.380s
sys	0m5.246s

for whatever reason, the program hung on 1+500e6, so just multiply 10.93 by 5
to get ~54.7s, or about 32x slower than C (and roughly 2x as fast as Ruby)
'''
print sum(range(int(1+100e6)))

File: sum.c
/*
$ time ./a.out
125000000250000000

real	0m1.703s
user	0m1.689s
sys	0m0.007s
*/
#include <stdio.h>

int main()
{
  long limit=500000000;
  unsigned long sum=0;
  for (long i=0; i<=limit; i++)
  {  sum+=i;
  }

  printf("%lu\n", sum);
  return 0;
}
